homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -133,12 +133,13 @@ def __init__(
 
     @callback
     def _update_attr(self, surepy_entity: SurepyEntity) -> None:
-        state = surepy_entity.raw_data()["status"]
-        self._attr_is_on = bool(state)
-        if state:
-            self._attr_extra_state_attributes = {
-                "device_rssi": f"{state['signal']['device_rssi']:.2f}",
-                "hub_rssi": f"{state['signal']['hub_rssi']:.2f}",
-            }
-        else:
-            self._attr_extra_state_attributes = {}
+        state = surepy_entity.raw_data().get("status", {})
+        online = bool(state.get("online", False))
+        self._attr_is_on = online
+        self._attr_extra_state_attributes = {}
+        if online:
+            value = state.get("device_rssi", {}).get(attr)
+            self._attr_extra_state_attributes["device_rssi"] = value ? f"{value:.2f}" : "Unknown"
```

Comment:
I really shouldn't review Python code while reading Ruby… Sorry!
```suggestion
            self._attr_extra_state_attributes["device_rssi"] = f"{value:.2f}" if value else "Unknown"
```

Reply:
Still not working :(
`Error while setting up surepetcare platform for binary_sensor: name 'attr' is not defined
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 382, in _async_setup_platform
    await asyncio.shield(awaitable)
  File "/config/custom_components/surepetcare/binary_sensor.py", line 44, in async_setup_entry
    entities.append(DeviceConnectivity(surepy_entity.id, coordinator))
                    ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/surepetcare/binary_sensor.py", line 130, in __init__
    super().__init__(surepetcare_id, coordinator)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/surepetcare/binary_sensor.py", line 62, in __init__
    super().__init__(surepetcare_id, coordinator)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/surepetcare/entity.py", line 45, in __init__
    self._update_attr(coordinator.data[surepetcare_id])
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/surepetcare/binary_sensor.py", line 141, in _update_attr
    value = state.get("device_rssi", {}).get(attr)
                                             ^^^^
NameError: name 'attr' is not defined. Did you mean: 'aiter'?`

Reply:
ah yes, this is from the "let's not do a loop" change, see https://github.com/home-assistant/core/pull/143286/files#r2053811360 and https://github.com/home-assistant/core/pull/143286/files#r2053846797